### PR TITLE
AT-3097: zendesk - zendesk_downloader_updated

### PR DESCRIPTION
**Pull request by tommy-watts**

### CODE REVIEW

File: pygyver/etl/zendesk.py [hunk 1]
```diff
@@ -149,28 +149,24 @@ def set_endpoint(self):
         The first call sets the endpoint based on values from the provided uri and params.
         Subsequent calls set the endpoint based on the returned `next_page` value from the previous call.
```
Comment (dankruse1) [inline pygyver/etl/zendesk.py:150]: should probably amend this

File: pygyver/etl/zendesk.py [hunk 2]
```diff
@@ -149,28 +149,24 @@ def set_endpoint(self):
...
         self.retry = 1
-        if self.page_number == 1:
+        if not self.endpoint:
             self.endpoint = self.base_url + self.uri + self.endpoint_params
             logging.info("Endpoint generated - {}".format(self.endpoint))
         else:
-            self.endpoint = self.data["next_page"]
+            next_endpoint = self.data.get("next_page", self.data.get("after_url"))
+            if next_endpoint:
+              self.endpoint = self.data.get("next_page", self.data.get("after_url"))
```
Comment (dankruse1) [inline pygyver/etl/zendesk.py:159]: self.endpoint =next.endpoint

**Review by dankruse1 — CHANGES_REQUESTED** [review]

File: pygyver/etl/zendesk.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ class ZendeskDownloader:
     Class to download data from a Zendesk API endpoint specified by params.
     """
 
-    def __init__(self, start_date=None, uri=None, sideload=None, per_page=500, pagination=True):
+    def __init__(self, start_date=None, uri=None, sideload=None, per_page=500, limit=None, pagination=True, base_url="https://madecom.zendesk.com/"):
```
Comment (jean-maldo) [inline pygyver/etl/zendesk.py:24]: `pagination=True` should be removed right? Since it's not being used anymore.

File: pygyver/etl/zendesk.py [hunk 4]
```diff
@@ -42,27 +42,28 @@ def __init__(self, start_date=None, uri=None, sideload=None, per_page=500, pagin
...
         self.user = "analytics@made.com/token"
-        self.base_url = "https://madecom.zendesk.com/"
+        self.base_url = base_url
         self.start_date = start_date
         self.sideload = sideload
         self.uri = uri
         self.per_page = per_page
-        self.pagination = pagination
+        self.limit = limit
+        # self.pagination = pagination
```
Comment (jean-maldo) [inline pygyver/etl/zendesk.py:51]: May as well delete rather than comment out

File: vars/build/default/default.env [hunk 5]
```diff
@@ -1,3 +1,3 @@
 PY_VERSION=3.8.1
 BASE_IMAGE_REPO=git@github.com:madedotcom/analytics-python-docker-base.git
-PYTHONPATH=/code/src/
\ No newline at end of file
+IMAGE_REGISTRY=gcr.io/copper-actor-127213
```
Comment (jean-maldo) [inline vars/build/default/default.env:3]: Add new line

File: vars/build/default/default.env [hunk 6]
```diff
@@ -1,3 +1,3 @@
 PY_VERSION=3.8.1
 BASE_IMAGE_REPO=git@github.com:madedotcom/analytics-python-docker-base.git
-PYTHONPATH=/code/src/
```
Comment (jean-maldo) [inline vars/build/default/default.env:3]: Reason for removing this?

File: vars/build/test/test.env [hunk 7]
```diff
@@ -1 +0,0 @@
-PROJECT_ROOT=/code/src/
```
Comment (jean-maldo) [inline vars/build/test/test.env:1]: Reason for removing this?

File: vars/env/default/default.env [hunk 8]
```diff
@@ -1 +1,3 @@
+# python path
 PYTHONPATH=/code/src/
+PROJECT_ROOT=/code/src/
```
Comment (jean-maldo) [inline vars/env/default/default.env:3]: Add new line

File: vars/env/local/README.md [hunk 9]
```diff
@@ -1,16 +0,0 @@
...
-- local.env
-
-## local.env ###
-
-```R
-GOOGLE_APPLICATION_CREDENTIALS=
-FACEBOOK_APPLICATION_CREDENTIALS=
-BIGQUERY_PROJECT=
-PROJECT_ROOT=
-```
```
Comment (jean-maldo) [inline vars/env/local/README.md:16]: This file shouldn't be deleted

File: vars/env/test/test.env [hunk 10]
```diff
@@ -1,7 +1,20 @@
+BIGQUERY_ACCESS_TOKEN_PATH=/code/access_token.json
 BIGQUERY_PROJECT=invertible-star-147708
 GCS_PROJECT=invertible-star-147708
-GCS_BUCKET=madecom-bi-sandbox
-PROJECT_ROOT=/code/src/
+GCS_BUCKET=madecom-ds-test
+GCS_AUTH_FILE=/code/access_token.json
 GOOGLE_APPLICATION_CREDENTIALS=/code/access_token.json
-GOODDATA_DOMAIN="https://fake-gooddata-project.com"
-GOODDATA_PROJECT="12345"
```
Comment (jean-maldo) [inline vars/env/test/test.env:7]: Should these be changed?

**Review by jean-maldo — DISMISSED** [review]

File: pygyver/etl/zendesk.py [hunk 11]
```diff
@@ -21,63 +21,63 @@ class ZendeskDownloader:
...
         """
         Construct all the necessary attributes for the ZendeskDownloader instance
         
         Args:
           uri (str): API endpoint.
-          start_date (str, optional): Timestamp format ('2018-12-31 22:45:32').
+          start_time (str, optional): Timestamp format ('2018-12-31 22:45:32').
           sideload (str, optional): Additional parameters seperated by &.
           per_page (int, optional): Size of page to fetch from the API.
           pagination (bool, optional): Flag for if the endpoint has pagination.
```
Comment (dankruse1) [inline pygyver/etl/zendesk.py:33]: delete

Comment (dankruse1) [reply pygyver/etl/zendesk.py:33]: and add limit

File: pygyver/etl/zendesk.py [hunk 12]
```diff
@@ -112,18 +112,23 @@ def set_uri(self):
...
     def set_per_page_param(self):
         """Format and append per_page param to params_list if `self.per_page` is not None
         """
-        if self.pagination:
+        if self.per_page:
           param = "per_page={}".format(self.per_page)
           self.params_list.append(param)
+
+    def set_limit_param(self):
+      if self.limit:
```
Comment (dankruse1) [inline pygyver/etl/zendesk.py:129]: add doc string

**Review by dankruse1 — CHANGES_REQUESTED** [review]

File: pygyver/etl/zendesk.py [hunk 13]
```diff
@@ -149,48 +156,21 @@ def set_endpoint(self):
...
+            if next_endpoint:
+              self.endpoint = next_endpoint
+            else:
+              self.finished = True
 
     def api_call(self):
         """ Make the API request and handle retry based response codes.
         
-        After each request update `self.count` with the number of records returned, `self.next_page` with the next_page endpoint provided in the response.
+        After each request update `self.endpoint` with the next_page endpoint provided in the response.
```
Comment (dankruse1) [inline pygyver/etl/zendesk.py:173]: could maybe use a generic term here or say next_page/limit

File: pygyver/etl/zendesk.py [hunk 14]
```diff
@@ -21,63 +21,63 @@ class ZendeskDownloader:
     Class to download data from a Zendesk API endpoint specified by params.
     """
 
-    def __init__(self, start_date=None, uri=None, sideload=None, per_page=500, pagination=True):
+    def __init__(self, start_time=None, uri=None, sideload=None, per_page=None, limit=None, base_url="https://madecom.zendesk.com/"):
         """
         Construct all the necessary attributes for the ZendeskDownloader instance
         
         Args:
           uri (str): API endpoint.
```
Comment (dankruse1) [inline pygyver/etl/zendesk.py:29]: add start_time

**Review by dankruse1 — CHANGES_REQUESTED** [review]

**Review by dankruse1 — APPROVED** [review]

**Pull request merged**